src/prefect/environments/execution/dask/k8s.py

```diff
@@ -81,7 +81,7 @@ def __init__(
         self,
         min_workers: int = 1,
         max_workers: int = 2,
-        work_stealing: bool = False,
+        work_stealing: bool = True,
```

Comment:
I didn't know this was the default, I'm pretty sure this explains #2958. Disabling this would mean that tasks assigned to one worker would never rebalance, and since autoscaling is used, all tasks would be assigned to the original single worker. +100 on changing this.